File: Freedom/微信/我/Setting/WXMineSettingViewController.m
//  TLMineSettingViewController.m
//  Freedom
//  Created by Super on 16/2/8.
#import "WXMineSettingViewController.h"
#import "WXAccountAndSafetyViewController.h"
#import "WXNewMessageSettingViewController.h"
#import "WXPrivacySettingViewController.h"
#import "WXCommonSettingViewController.h"
#import "WXHelpAndFeedbackViewController.h"
#import "WXAboutViewController.h"
#import "WXModes.h"
typedef void (^ CompleteBlock)(NSMutableArray *data);
@interface WXSettingHelper : NSObject
@property (nonatomic, strong) NSMutableArray *mineSettingData;
@end
@implementation WXSettingHelper
- (id) init{
    if (self = [super init]) {
        self.mineSettingData = [[NSMutableArray alloc] init];
        [self p_initTestData];
    }
    return self;
}
- (void) p_initTestData{
    WXSettingItem *item1 = TLCreateSettingItem(@"账号与安全");
    if (/* DISABLES CODE */ (1)) {
        item1.subTitle = @"已保护";
        item1.rightImagePath = @"u_protectHL";
    }else {
        item1.subTitle = @"未保护";
        item1.rightImagePath = @"u_protect";
    }
    WXSettingGroup *group1 = TLCreateSettingGroup(nil, nil, @[item1]);
    
    WXSettingItem *item2 = TLCreateSettingItem(@"新消息通知");
    WXSettingItem *item3 = TLCreateSettingItem(@"隐私");
    WXSettingItem *item4 = TLCreateSettingItem(@"通用");
    WXSettingGroup *group2 = TLCreateSettingGroup(nil, nil, (@[item2, item3, item4]));
    
    WXSettingItem *item5 = TLCreateSettingItem(@"帮助与反馈");
    WXSettingItem *item6 = TLCreateSettingItem(@"关于微信");
    WXSettingGroup *group3 = TLCreateSettingGroup(nil, nil, (@[item5, item6]));
    
    WXSettingItem *item7 = TLCreateSettingItem(@"退出登录");
    item7.type = TLSettingItemTypeTitleButton;
    WXSettingGroup *group4 = TLCreateSettingGroup(nil, nil, @[item7]);
    
    [self.mineSettingData addObjectsFromArray:@[group1, group2, group3, group4]];
}
@end
@interface WXMineSettingViewController ()
@property (nonatomic, strong) WXSettingHelper *helper;
@end
@implementation WXMineSettingViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    [self.navigationItem setTitle:@"设置"];
    
    self.helper = [[WXSettingHelper alloc] init];
    self.data = self.helper.mineSettingData;
}
#pragma mark - 
//MARK: UITableViewDelegate
- (void) tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    WXSettingItem *item = [self.data[indexPath.section] objectAtIndex:indexPath.row];
    if ([item.title isEqualToString:@"账号与安全"]) {
        WXAccountAndSafetyViewController *accountAndSafetyVC = [[WXAccountAndSafetyViewController alloc] init];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:accountAndSafetyVC animated:YES];
    }else if ([item.title isEqualToString:@"新消息通知"]) {
        WXNewMessageSettingViewController *newMessageSettingVC = [[WXNewMessageSettingViewController alloc] init];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:newMessageSettingVC animated:YES];
    }else if ([item.title isEqualToString:@"隐私"]) {
        WXPrivacySettingViewController *privacySettingVC = [[WXPrivacySettingViewController alloc] init];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:privacySettingVC animated:YES];
    }else if ([item.title isEqualToString:@"通用"]) {
        WXCommonSettingViewController *commonSettingVC = [[WXCommonSettingViewController alloc] init];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:commonSettingVC animated:YES];
    }else if ([item.title isEqualToString:@"帮助与反馈"]) {
        WXHelpAndFeedbackViewController *helpAndFeedbackVC = [[WXHelpAndFeedbackViewController alloc] init];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:helpAndFeedbackVC animated:YES];
    }else if ([item.title isEqualToString:@"关于微信"]) {
        WXAboutViewController *aboutVC = [[WXAboutViewController alloc] init];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:aboutVC animated:YES];
    }else if ([item.title isEqualToString:@"退出登录"]) {
        [self showAlerWithtitle:@"退出后不会删除任何历史数据，下次登录依然可以使用本账号。" message:nil style:UIAlertControllerStyleActionSheet ac1:^UIAlertAction *{
            return [UIAlertAction actionWithTitle:@"退出登录" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
            }];
        } ac2:^UIAlertAction *{
            return [UIAlertAction actionWithTitle:@"取消" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {

            }];
        } ac3:nil completion:nil];
    }
    [super tableView:tableView didSelectRowAtIndexPath:indexPath];
}
@end

File: Freedom/微信/我/Setting/WXMineSettingViewController.swift
//
//  WXMineSettingViewController.swift
//  Freedom

import Foundation
typealias CompleteBlock = ([AnyHashable]) -> Void
class WXSettingHelper: NSObject {
    var mineSettingData: [WXSettingGroup] = []
    override init() {
        super.init()
        let item1 = WXSettingItem.createItem(withTitle: ("账号与安全"))
        if (1) != 0 {
            item1.subTitle = "已保护"
            item1.rightImagePath = "u_protectHL"
        } else {
            item1.subTitle = "未保护"
            item1.rightImagePath = "u_protect"
        }
        let group1: WXSettingGroup = TLCreateSettingGroup(nil, nil, [item1])
        let item2 = WXSettingItem.createItem(withTitle: ("新消息通知"))
        let item3 = WXSettingItem.createItem(withTitle: ("隐私"))
        let item4 = WXSettingItem.createItem(withTitle: ("通用"))
        let group2: WXSettingGroup = TLCreateSettingGroup(nil, nil, ([item2, item3, item4]))
        let item5 = WXSettingItem.createItem(withTitle: ("帮助与反馈"))
        let item6 = WXSettingItem.createItem(withTitle: ("关于微信"))
        let group3: WXSettingGroup = TLCreateSettingGroup(nil, nil, ([item5, item6]))
        let item7 = WXSettingItem.createItem(withTitle: ("退出登录"))
        item7.type = .titleButton
        let group4: WXSettingGroup = TLCreateSettingGroup(nil, nil, [item7])
        mineSettingData.append(contentsOf: [group1, group2, group3, group4])
    }
}
class WXMineSettingViewController: WXSettingViewController {
    var helper = WXSettingHelper()
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "设置"
        data = helper.mineSettingData
    }
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let item = data[indexPath.section].items[indexPath.row]
        if (item.title == "账号与安全") {
            let accountAndSafetyVC = WXAccountAndSafetyViewController()
            hidesBottomBarWhenPushed = true
            navigationController?.pushViewController(accountAndSafetyVC, animated: true)
        } else if (item.title == "新消息通知") {
//            let newMessageSettingVC = WXNewMessageSettingViewController()
//            hidesBottomBarWhenPushed = true
//            navigationController?.pushViewController(newMessageSettingVC, animated: true)
        } else if (item.title == "隐私") {
//            let privacySettingVC = WXPrivacySettingViewController()
//            hidesBottomBarWhenPushed = true
//            navigationController?.pushViewController(privacySettingVC, animated: true)
        } else if (item.title == "通用") {
            let commonSettingVC = WXCommonSettingViewController()
            hidesBottomBarWhenPushed = true
            navigationController?.pushViewController(commonSettingVC, animated: true)
        } else if (item.title == "帮助与反馈") {
            let helpAndFeedbackVC = WXHelpAndFeedbackViewController()
            hidesBottomBarWhenPushed = true
            navigationController?.pushViewController(helpAndFeedbackVC, animated: true)
        } else if (item.title == "关于微信") {
            let aboutVC = WXAboutViewController()
            hidesBottomBarWhenPushed = true
            navigationController?.pushViewController(aboutVC, animated: true)
        } else if (item.title == "退出登录") {
            let alert = UIAlertController("退出后不会删除任何历史数据，下次登录依然可以使用本账号。", "", T1: "退出登录", T2: "取消", confirm1: {

            }, confirm2: {

            })
            self.present(alert, animated: true, completion: nil)
        }
        super.tableView(tableView, didSelectRowAt: indexPath)
    }
}
